model/host/hostutil.go

```diff
@@ -433,7 +433,22 @@ func (h *Host) GenerateUserDataProvisioningScript(ctx context.Context, settings
 			if err != nil {
 				return "", errors.Wrap(err, "constructing Jasper command to fetch task data")
 			}
-			postFetchClient += " && " + getTaskDataCmd
+			if !h.ProvisionOptions.UseOAuth {
+				// The legacy approach is to run `evergreen fetch` directly here with
+				// static credentials.
+				postFetchClient += " && " + getTaskDataCmd
+			} else {
+				// Escape single quotes in the command.
+				// This is done by closing the single quoted string, starting a
+				// double quoted string, having it's content be a single quote, then
+				// closing the double quoted string, and reopening the original
+				// single quoted string.
+				getTaskDataCmd := strings.ReplaceAll(getTaskDataCmd, "'", `'"'"'`)
```

Comment:
Will this leak the githubAppToken and moduleToken to that file?

Reply:
Yes, but I don't think it's a concern. The clone has the token referenced regardless. Removing the token from the file doesn't improve our security profile. If we fetch it on-demand, I think we actually worse our security profile because a malicious user could just run however we fetch it on-demand at any time.

Reply:
That's fair, but It think this improvement is out of scope for this ticket/project. I could split a ticket to improve this?

Reply:
For expiring tokens, I think we can just document it instead of improving it for now. 
For the concern about the token being left behind, because these are read-only tokens, I'm okay with splitting a ticket and leaving it out of the epic for now.